### ISSUE TITLE
How to get all probabilities for a class?

### ISSUE THREAD

**Issue by baglanaitu** [ultralytics/ultralytics#374]
### Search before asking

- [x] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have the similar question which was asked in yolov5: https://github.com/ultralytics/yolov5/issues/7183.

If I have 4 classes in my dataset (for example: A, B, C, D). Let's say, yolov8 detects one object in an image and gives label of one class (for example, A). How can I get probabilities for that object being other class (B, C, D)? At the end, their sum must be equal to 1.

**Comment by baglanaitu** [comment]
Hi @AbdelsalamHaa ! Thank you for your answer! Do you know by any chance how I can apply softmax in yolov8? 
I tried this https://github.com/ultralytics/yolov5/issues/584 with yolov7 for 50 epochs, and model almost learnt nothing and there were no detections in the output (usually with default SiLU function it should learn and detect some objects even with 50 epochs). I don't think more epochs can solve the problem. Any suggestion would help! Thank you!

**Comment by AbdelsalamHaa** [comment]
@baglanaitu I'm not really sure where exactly u need to modify it if u want to be added during the training. But maybe u can just apply it on the output only. That could probably work too.

**Comment by prince0310** [comment]
@baglanaitu cheak out  `/ultralytics/ultralytics/yolo/engine/trainer.py` line no 554 for optimizer tuning
